### PR TITLE
Fix CDP cookie matching algorithm

### LINKED ISSUE #5656: Cypress incorrectly validates domain of cookies in 3.5.0
### Current behavior:

Cookies are failing to set for subdomains during `cy.request`.
E.g. we call `/auth` of `auth.test.server` and it returns `token` cookie with `.test.server`. In this case cookie is not set.

### Desired behavior:

Cookie is set in above example.

### Steps to reproduce: (app code and test code)

Look at [packages/server/lib/request.coffee#setCookiesOnBrowser](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/request.coffee#L449):

```
return if not tough.domainMatch(cookie.domain, parsedUrl.hostname)
```

And at [`tough-cookie` documentation](https://www.npmjs.com/package/tough-cookie#domainmatchstrdomstrcanonicalizetrue):

```
domainMatch(str,domStr[,canonicalize=true])
Answers "does this real domain match the domain in a cookie?". The str is the "current" domain-name and the domStr is the "cookie" domain-name. Matches according to RFC6265 Section 5.1.3, but it helps to think of it as a "suffix match".
```

So it should be:

```
return if not tough.domainMatch(parsedUrl.hostname, cookie.domain)
```

### Versions

`>=3.5.0`

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #5656

### Additional details

- previously, we were using `tough.cookieMatch` to filter cookies: https://github.com/cypress-io/cypress/blob/d56c551b0bb77a838da8de15bff137f1d96f9d47/packages/server/lib/browsers/cdp_automation.ts#L23-L37
- however, this is *incorrect*
	- Cypress was designed around the internal 'cookie filter' having the same structure as the filter for WebExtensions cookies.getAll: https://developer.chrome.com/extensions/cookies#method-getAll
	- the WebExtensions filter uses `domain` like this: "Restricts the retrieved cookies to those whose domains match or are subdomains of this one."
	- we were using tough.domainMatch, which matches like an actual browser, instead

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub?

### CODE REVIEW

File: packages/server/lib/browsers/cdp_automation.ts [hunk 1]
```diff
@@ -18,18 +17,29 @@ interface CyCookie {
   httpOnly: boolean
 }
 
+// Cypress uses the webextension-style filtering
+// https://developer.chrome.com/extensions/cookies#method-getAll
+type CyCookieFilter = chrome.cookies.GetAllDetails
```
Comment (brian-mann) [inline packages/server/lib/browsers/cdp_automation.ts]: why not import the `chrome` types at the top vs putting them in the `compilerOptions`? I feel like this comes out of nowhere

Comment (flotwig) [reply packages/server/lib/browsers/cdp_automation.ts]: if i do `import 'chrome'`, build fails

if i do `import chrome from 'chrome'`, it says 'chrome' is not a module

i can dig into why `import 'chrome'` makes the build fail if you'd like, i'm not sure this is the best way either

Comment (brian-mann) [reply packages/server/lib/browsers/cdp_automation.ts]: ask @tgriesser in #typescript channel

Comment (brian-mann) [reply packages/server/lib/browsers/cdp_automation.ts]: nvm he's on his way to the airport - @tgriesser or @thlorenz can you explain why this is necessary whenever you have time?

**Pull request merged**